catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -402,6 +400,42 @@ private static Subject getSubject() {
     return (Subject) SecurityUtils.getSubject();
   }
 
+  private Stream<Metacard> getAllTemplateMetacardsByTag(String tag) {
+    final Map<String, Result> systemTemplateMetacards = getSystemTemplateMetacardsByTag(tag);
+    final Map<String, Result> myTemplateMetacards = util.getMetacardsByTag(tag);
```

Comment:
Could these 2 be combined into a single OR query for system templates and the user's templates?

Reply:
Technically no because the subject is different on each request, but @stustison and I might have found a separate solution so that these changes to search forms aren't necessary to begin with.